src/coprocessor/endpoint.rs

```diff
@@ -953,9 +957,20 @@ mod tests {
         let mut end_point = Host::new(engine, worker.scheduler(), &cfg, pd_worker.scheduler());
         end_point.max_running_task_count = 1;
         worker.start(end_point).unwrap();
-        let (tx, rx) = mpsc::channel();
+        let (tx, rx) = futures_mpsc::channel(150);
         for pos in 0..30 * 4 {
+            let (delayed_tx, delayed_rx) = futures_mpsc::channel(1);
             let tx = tx.clone();
+            thread::spawn(move || {
+                let _ = tx.send_all(
+                    delayed_rx
+                        .map_err(|_| unreachable!() /* See futures-rs #401 */)
+                        .map(|v| {
+                            thread::sleep(Duration::from_millis(100));
+                            v
+                        }),
+                ).wait();
```

Comment:
@BusyJay I would like to know why `tx.send_all(..).wait()` will get `SendError` here. As I understand, the channel size is larger than possible requests and it will never full?

Reply:
I guess that's because `rx` get dropped earlier than `tx`.